src/utils/polyfillEnvironment.js

```diff
@@ -32,11 +32,12 @@ if (!global.self) {
   global.self = global; /* eslint-disable-line */
 }
 
-require('InitializeCore'); // eslint-disable-line import/no-unresolved
+require('react-native/Libraries/Core/InitializeCore');
 
 require('../hot/client/importScriptsPolyfill');
 
-const scriptURL = require('react-native').NativeModules.SourceCode.scriptURL; // eslint-disable-line import/no-unresolved
+const scriptURL = require('react-native/Libraries/BatchedBridge/NativeModules')
```

Comment:
there's an import for Native modules, so you can do
```js
const { NativeModules } = require('react-native');
const { scriptURL } = NativeModules.SourceCode;
```

Reply:
yea, there is a trick i'm doing to prevent the use of `react-native` main entry point. I'll show you in the code example I'm using but the tree shaking doesn't work properly when all modules are exported from this entry point. The getters that do the dynamic require break the tree shaking, so I use webpack to alias that. That is why i'm reaching into the `react-native` library code to find the NativeModules.

Reply:
I'm curious how you import regular modules like `View` or `Text`. 

It's worth noting, that `react-native` is official public API and paths like `react-native/Libraries/BatchedBridge/NativeModules` may change in future as it's an implementation detail – I'm a bit concerned about that.

Reply:
Yea, I used a babel plugin that I found to rewrite module imports, I suppose I could force that to be used for haul as well so that it gets rewritten. That was I can use NativeModules in the way that you suggested.

The babel-plugin transforms the imports that look like this:
```
import { Text } from 'react-native
```

Transforming them to:
```
import Text from 'react-native/Libraries/Text/Text'
```

So I could change my commit to use that directly. I provide my own traversal which looks at `react-native-implementation.js` and finds all the exported symbols and the paths to the requires.